py_cui/control_widgets/slider.py

```diff
@@ -153,20 +153,25 @@ def _custom_draw_border(self, start_y: int):
         renderer = self.get_renderer()
         ui_element = self
 
+        renderer.set_color_mode(ui_element.get_border_color())
+
         if ui_element.is_selected():
             renderer._set_bold()
+            renderer._draw_border_top(ui_element, start_y, False)
 
-        renderer.set_color_mode(ui_element.get_border_color())
+            renderer.draw_text(self, content, start_y + 1, selected=ui_element.is_selected(), bordered=True)
+            renderer._set_bold()
+
+            renderer._draw_border_bottom(ui_element, start_y + 2)
+            renderer._unset_bold()
+        else:
+            renderer._draw_border_top(ui_element, start_y, False)
+            renderer.draw_text(self, content, start_y + 1, selected=ui_element.is_selected(), bordered=True)
+            renderer._draw_border_bottom(ui_element, start_y + 2)
```

Comment:
I think this logic could be simplified, something like:

```
if ui_element.is_selected():
    renderer._set_bold()

DO THE DRAWING HERE

if ui_element.is_selected():
    renderer._unset_bold()
```

This way the drawing logic doesn't have to be repeated. Also, since this method isn't just drawing the border now, maybe it should be renamed to `_custom_draw_with_border` instead?

Reply:
That was what I did, but I didn't. Because the same logic is also implemented in ```draw_text``` so it unsets the cursor after it.

https://github.com/jwlodek/py_cui/blob/671e43bf6f8655ae158497298176446ab41338c3/py_cui/renderer.py#L402-L413

If I set ```selected=False``` on ```draw_text``` to workaround it:

https://github.com/jwlodek/py_cui/blob/671e43bf6f8655ae158497298176446ab41338c3/py_cui/renderer.py#L371

This line will disable it afterward.

Then I need one more ```if``` block to set the bold again. 2 of those are good to read, but 3 harms readability.

```python
    if ui_element.is_selected():
        renderer._set_bold()
        
    renderer._draw_border_top(ui_element, start_y, False)
    renderer.draw_text(self, content, start_y + 1, selected=ui_element.is_selected(), bordered=True)
    
    if ui_element.is_selected():
        renderer._set_bold()

    renderer._draw_border_bottom(ui_element, start_y + 2)

    if ui_element.is_selected():
            renderer._unset_bold()
```

And that's why I simply made a branching because that is more readable because ironically this reduces repeating those ```if``` blocks.

About renaming, I just found out that borderless version is not getting bold when selected, I'll update it altogether.

Reply:
Okay yeah I forgot that bold is enabled/disabled in the draw_text function. I guess your solution is fine for now.

Reply:
Lemme know if there's something to help. I lack a fancy skills, though-